features/device-config/api/src/main/java/org/opennms/features/deviceconfig/service/DeviceConfigService.java

```diff
@@ -38,20 +39,28 @@
      *
      * @param ipAddress  specific IpAddress for which we need to fetch device config.
      * @param location   specific minion location at which we need to fetch device config.
-     * @param configType  configType whether it is Default or Running.
+     * @param service    name of the bound service.
      * @throws IOException
      */
-    void triggerConfigBackup(String ipAddress, String location, String configType) throws IOException;
+    void triggerConfigBackup(String ipAddress, String location, String service) throws IOException;
 
     /**
      * Get device config for the given ipAddress at given location.
      *
      * @param ipAddress  specific IpAddress for which we need to fetch device config.
      * @param location   specific minion location at which we need to fetch device config.
-     * @param configType configType whether it is Default or Running.
+     * @param service    name of the bound service.
      * @param timeout    timeout in milliseconds for retrieving device config
      * @throws IOException
      * @return
      */
-    CompletableFuture<byte[]> getDeviceConfig(String ipAddress, String location, String configType, int timeout) throws IOException;
+    CompletableFuture<byte[]> getDeviceConfig(String ipAddress, String location, String service, int timeout) throws IOException;
+
+    List<RetrivalConfiguration> getRetrivalConfigurations(String ipAddress, String location);
+
+    interface RetrivalConfiguration {
```

Comment:
Here and elsewhere: `Retrival` -> `Retrieval`

Reply:
fixed